sidekiq_metrics.go

```diff
@@ -0,0 +1,149 @@
+//
+// Copyright 2018, Sander van Harmelen
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package gitlab
+
+import "time"
+
+// SidekiqService handles communication with the application SettingsService
+// related methods of the GitLab API.
+//
+// GitLab API docs: https://docs.gitlab.com/ce/api/settings.html
+type SidekiqService struct {
+	client *Client
+}
+
+// QueueMetricsStat represents the QueueMetrics embedded struct.
+type QueueMetricsStat struct {
+	Backlog int `json:"backlog"`
+	Latency int `json:"latency"`
+}
+
+// QueueMetrics represents the GitLab sidekiq queue metrics.
+//
+// GitLab API docs: https://docs.gitlab.com/ce/api/sidekiq_metrics.html#get-the-current-queue-metrics
+type QueueMetrics struct {
+	Queues map[string]QueueMetricsStat `json:"queues"`
+}
```

Comment:
Using a map is fine here, but I would suggest to just define the `QueueMetricsStat` fields inline here...

```
Queues map[string]struct {
  Backlog int `json:"backlog"`
  Latency int `json:"latency"`
} `json:"queues"`
```